cpp/src/parquet/reader_test.cc

```diff
@@ -502,6 +502,66 @@ TEST_F(TestAllTypesPlain, ColumnSelectionOutOfRange) {
   ASSERT_THROW(printer2.DebugPrint(ss, columns), ParquetException);
 }
 
+// Tests that read_dense_for_nullable is passed down to the record
+// reader. The functionality of read_dense_for_nullable is tested
+// elsewhere.
+TEST(TestFileReader, RecordReaderReadDenseForNullable) {
+  // Default is false.
```

Comment:
Can we use a vector of ReaderProperties to reduce repeated code? For example,
```cpp
std::vector<ReaderProperties> reader_properties(3);
reader_properties[1].enable_read_dense_for_nullable();
reader_properties[2].enable_read_dense_for_nullable();
reader_properties[2].disable_read_dense_for_nullable();
for (const auto& reader_props : reader_properties) {
    std::unique_ptr<ParquetFileReader> file_reader = ParquetFileReader::OpenFile(
        alltypes_plain(), /* memory_map = */ false, reader_props);
    std::shared_ptr<RowGroupReader> group = file_reader->RowGroup(0);
    std::shared_ptr<internal::RecordReader> col_record_reader = group->RecordReader(0);
    ASSERT_EQ(reader_props.read_dense_for_nullable(), 
              col_record_reader->read_dense_for_nullable());
}
```

Reply:
Done.